project2/src/Page/Home/HomePage.jsx

```diff
@@ -16,6 +17,20 @@ function HomePage() {
     const [categories, setCategories] = useState([]);
     const [mostViewProducts, setMostViewProducts] = useState([]);
 
+    useEffect(() => {
+        const userId = JSON.parse(localStorage.getItem("user")).user.id;
+        if (localStorage.getItem("cart")) {
```

Comment:
```suggestion
        const cart = localStorage.getItem("cart");
        if (cart.length) {
```
Xong đó thì check userId ở trong block if này nhé